posthog/clickhouse/cluster.py

```diff
@@ -422,3 +417,26 @@ def __convert_lightweight_delete_to_mutation_command(self) -> str:
             raise ValueError(f"Invalid DELETE command format: {self.command}")
         where_clause = self.command.strip()[match.end() :]
         return f"UPDATE _row_exists = 0 WHERE {where_clause}"
+
+    def run_on_shards(self, cluster: ClickhouseCluster, shards: Iterable[int] | None = None) -> None:
+        """
+        Enqueue (or find) this mutation on one host in each shard, and then block until the mutation is complete on all
+        hosts within the affected shards.
+        """
+        if shards is not None:
+            shard_host_mutations = cluster.map_any_host_in_shards({shard: self.enqueue for shard in shards})
+        else:
+            shard_host_mutations = cluster.map_one_host_per_shard(self.enqueue)
+
+        # XXX: need to convert the `shard_num` of type `int | None` to `int` to appease the type checker -- but nothing
+        # should have actually been filtered out, since we're using the cluster shard functions for targeting
+        shard_mutations = {
+            host.shard_num: mutations
+            for host, mutations in shard_host_mutations.result().items()
+            if host.shard_num is not None
+        }
+        assert len(shard_mutations) == len(shard_host_mutations)
```

Comment:
This is kludgey, will put this on my list of things to improve when/if I get around to cleaning up with `ClickhouseCluster` interface, there are already a number of other things I'd like to fix or improve